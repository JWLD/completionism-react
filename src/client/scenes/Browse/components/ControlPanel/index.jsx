import React from 'react'

const ControlPanel = () => <h1>I am a control panel yay!</h1>

export default ControlPanel
