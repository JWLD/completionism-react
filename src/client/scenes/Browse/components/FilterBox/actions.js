import * as ACTIONS from 'constants/action_types'

// ACTION CREATORS

export const setFilter = value => ({
  type: ACTIONS.SET_FILTER_VALUE,
  value
})

// THUNKS

export const resetFilter = () => dispatch => {
  dispatch(setFilter(''))
}
