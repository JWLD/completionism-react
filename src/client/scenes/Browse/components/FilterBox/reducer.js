import * as ACTIONS from 'constants/action_types'

export const initialState = {
  value: ''
}

const filterBoxReducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.SET_FILTER_VALUE:
      return { ...state, value: action.value }
    default:
      return state
  }
}

export default filterBoxReducer
