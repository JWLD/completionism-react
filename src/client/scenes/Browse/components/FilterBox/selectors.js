export const filterValueSelector = state => state.browse.filterBox.value
