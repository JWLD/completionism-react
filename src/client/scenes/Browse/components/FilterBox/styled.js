import styled from 'styled-components'
import { FaTimesCircle } from 'react-icons/lib/fa'

import { mixins, vars } from 'style'

export const FilterInput = styled.input`
  ${mixins.browseItem};

  padding: 1rem;
  background-color: #eee;
  font: 2.5rem DinMC;
`

export const ClearInputBtn = styled(FaTimesCircle)`
  display: ${props => !props.active && 'none'};
  position: absolute;
  right: 0;
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  margin-right: ${vars.itemListPadding}rem;
  padding: 0.75rem;
  color: gray;
  background-color: transparent;
  font-size: 3.25rem;

  :hover {
    cursor: pointer;
  }
`
