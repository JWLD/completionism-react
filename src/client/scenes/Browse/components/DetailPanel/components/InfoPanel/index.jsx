import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import { NPC_RENDER_URLS } from 'constants/urls'
import { itemDataSelector } from 'InfoPanel/selectors'

import * as SC from 'InfoPanel/styled'

const InfoPanel = ({ itemData }) => {
  const imageUrl =
    itemData.displayId && `${NPC_RENDER_URLS.zoom}${itemData.displayId}.jpg`

  return (
    <SC.InfoPanel>
      <SC.ItemName>{itemData.name || 'Select An Item'}</SC.ItemName>
      <SC.ItemImage imageUrl={imageUrl} />
      <SC.WowheadLink
        href={`http://www.wowhead.com/spell=${itemData.id}`}
        target="_blank">
        View on Wowhead
      </SC.WowheadLink>
    </SC.InfoPanel>
  )
}

InfoPanel.propTypes = {
  itemData: PropTypes.object.isRequired
}

const mapStateToProps = (state, props) => ({
  itemData: itemDataSelector(state, props)
})

const ReduxComponent = connect(mapStateToProps)(InfoPanel)

export default withRouter(ReduxComponent)
