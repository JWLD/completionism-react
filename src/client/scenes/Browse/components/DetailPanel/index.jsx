import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import { controlPanelSelector, itemDataSelector } from 'DetailPanel/selectors'
import { NPC_RENDER_URLS } from 'constants/urls'
import { toggleControlPanel } from 'DetailPanel/actions'

import * as SC from 'DetailPanel/styled'

export const DetailPanel = props => {
  const imageUrl = props.itemData.displayId
    ? `${NPC_RENDER_URLS.zoom}${props.itemData.displayId}.jpg`
    : null

  const title = props.controlPanelIsActive
    ? 'Control Panel'
    : props.itemData.name || 'Select An Item'

  const content = props.controlPanelIsActive ? null : (
    <SC.ItemImage imageUrl={imageUrl} />
  )

  return (
    <SC.DetailPanel>
      <SC.TopBar>
        <SC.Title>{title}</SC.Title>
        <SC.ToggleWrap onClick={() => props.toggleControlPanel()}>
          <SC.CogIcon active={props.controlPanelIsActive} />
          <SC.InfoIcon active={!props.controlPanelIsActive} />
        </SC.ToggleWrap>
      </SC.TopBar>
      {content}
    </SC.DetailPanel>
  )
}

DetailPanel.propTypes = {
  controlPanelIsActive: PropTypes.bool.isRequired,
  itemData: PropTypes.object.isRequired,
  toggleControlPanel: PropTypes.func.isRequired
}

const mapStateToProps = (state, props) => ({
  controlPanelIsActive: controlPanelSelector(state),
  itemData: itemDataSelector(state, props)
})

const mapDispatchToProps = {
  toggleControlPanel
}

const reduxComponent = connect(mapStateToProps, mapDispatchToProps)(DetailPanel)

export default withRouter(reduxComponent)
