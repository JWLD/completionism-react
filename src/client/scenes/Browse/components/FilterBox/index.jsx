import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { resetFilter, setFilter } from 'FilterBox/actions'

import * as SC from 'FilterBox/styled'

export const FilterBox = props => (
  <SC.FilterWrap>
    <SC.FilterInput
      onChange={e => props.setFilter(e.target.value)}
      value={props.filterValue}
      placeholder="Filter"
    />
    <SC.ClearInputBtn
      active={props.filterValue}
      onClick={() => props.resetFilter()}
    />
  </SC.FilterWrap>
)

FilterBox.propTypes = {
  filterValue: PropTypes.string.isRequired,
  resetFilter: PropTypes.func.isRequired,
  setFilter: PropTypes.func.isRequired
}

export const mapStateToProps = state => ({
  filterValue: state.browse.filter
})

const mapDispatchToProps = {
  resetFilter,
  setFilter
}

export default connect(mapStateToProps, mapDispatchToProps)(FilterBox)
