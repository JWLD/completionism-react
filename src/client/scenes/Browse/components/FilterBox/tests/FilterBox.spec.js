import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { FilterBox, mapStateToProps } from 'scenes/Browse/components/FilterBox'
import * as MOCKS from 'fixtures'

Enzyme.configure({ adapter: new Adapter() })

const mockResetFilterFunc = jest.fn()
const mockSetFilterFunc = jest.fn()

let component = null
let mockProps = {}

beforeEach(() => {
  mockProps = {
    filterValue: '',
    resetFilter: mockResetFilterFunc,
    setFilter: mockSetFilterFunc
  }

  jest.resetAllMocks()

  component = shallow(<FilterBox {...mockProps} />)
})

describe('<FilterBox />', () => {
  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })
})

describe('#setFilter', () => {
  let inputComponent = null

  beforeAll(() => {
    inputComponent = component.find('FilterInput')
  })

  it('is called when user types in input field', () => {
    inputComponent.simulate('change', { target: { value: '' } })

    expect(mockSetFilterFunc).toHaveBeenCalledTimes(1)
  })

  it('is called with the correct value', () => {
    const inputValue = 'foo'
    inputComponent.simulate('change', { target: { value: inputValue } })

    expect(mockSetFilterFunc).toHaveBeenCalledWith(inputValue)
  })
})

describe('#resetFilter', () => {
  it('is called when user clicks on clear input button', () => {
    const buttonComponent = component.find('ClearInputBtn')
    buttonComponent.simulate('click')

    expect(mockResetFilterFunc).toHaveBeenCalledTimes(1)
  })
})

describe('#mapStateToProps', () => {
  it('extracts the required data from state', () => {
    const result = mapStateToProps(MOCKS.INITIAL_STATE, mockProps)

    expect(Object.keys(result)).toHaveLength(1)
    expect(result).toHaveProperty('filterValue')
  })
})
