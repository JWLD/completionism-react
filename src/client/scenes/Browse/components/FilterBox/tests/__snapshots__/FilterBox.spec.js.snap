// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterBox /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterBox
    filterValue=""
    resetFilter={[MockFunction]}
    setFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FilterInput
          onChange={[Function]}
          placeholder="Filter"
          value=""
        />,
        <ClearInputBtn
          active=""
          onClick={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Filter",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FilterInput
            onChange={[Function]}
            placeholder="Filter"
            value=""
          />,
          <ClearInputBtn
            active=""
            onClick={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Filter",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
